code/modules/augment/active/hudimplant.dm

```diff
@@ -0,0 +1,60 @@
+/obj/item/organ/internal/augment/active/hud
+	name = "integrated HUD"
+	desc = "A small implantable heads-up display."
+	icon_state = "booster"
+
+	action_button_name = "Toggle integrated HUD"
+	allowed_organs = list(BP_AUGMENT_HEAD)
+
+	material = /decl/material/solid/metal/steel
+	matter = list(
+		/decl/material/solid/glass = MATTER_AMOUNT_REINFORCEMENT
+	)
+
+	origin_tech = "{'magnets':2}"
+
+	var/active = FALSE
+
+/obj/item/organ/internal/augment/active/hud/Process()
+	..()
+	if (owner && active)
+		if (allowed(owner))
+			callback_hud_to_process(owner, 1)
```

Comment:
Not a huge fan of this implementation: it's fragile, can have ordering problems, and doesn't update the mob's hud vars properly. I feel a better way to do this is to have mobs look through their organs for hud flags and add them before updating huds in Life. This can then set whether it has a hud flag (though I don't think the `allowed` check makes any sense).